app-text/booksorg/patches/booksorg-0.3.patchset

```diff
@@ -0,0 +1,275 @@
+From 5d7fe1ede47e9f2f49c2b161050b13762e7dd447 Mon Sep 17 00:00:00 2001
+From: mazbrili <mazbrili@yahoo.com>
+Date: Sat, 20 Oct 2018 18:56:36 +0700
+Subject: apply change
+
+
+diff --git a/bookinfo.cpp b/bookinfo.cpp
+old mode 100644
+new mode 100755
+index f75bcaa..1c69276
+--- a/bookinfo.cpp
++++ b/bookinfo.cpp
+@@ -20,7 +20,7 @@
+  */
+ #include "bookinfo.h"
+ #if QT_VERSION >= 0x050000
+-#include <poppler-qt5.h>
++#include <poppler/qt5/poppler-qt5.h>
+ #else
+ #include <poppler-qt4.h>
+ #endif
+diff --git a/bookinfo.h b/bookinfo.h
+old mode 100644
+new mode 100755
+index 445a9a7..0dc6f75
+--- a/bookinfo.h
++++ b/bookinfo.h
+@@ -23,7 +23,7 @@
+ #define BOOKINFO_H
+ #include <QString>
+ #if QT_VERSION >= 0x050000
+-#include <poppler-qt5.h>
++#include <poppler/qt5/poppler-qt5.h>
+ #else
+ #include <poppler-qt4.h>
+ #endif
+diff --git a/booksorganizer.pro b/booksorganizer.pro
+old mode 100644
+new mode 100755
+index 2162d85..6cf9cb0
+--- a/booksorganizer.pro
++++ b/booksorganizer.pro
+@@ -27,7 +27,10 @@ win32{
+     LIBS         += -L poppler-win\bin -lpoppler-qt5
+     RC_FILE = myapp.rc
+ #    QT +=winextras
+-}else{
++}
++haiku{
++	LIBS += -lpoppler -lpoppler-qt5
++	}else{
+ 
+         equals(QT_MAJOR_VERSION, 5) {
+           INCLUDEPATH  += /usr/include/poppler/qt5
+diff --git a/documentview.h b/documentview.h
+old mode 100644
+new mode 100755
+index dd911c0..7e68d96
+--- a/documentview.h
++++ b/documentview.h
+@@ -33,7 +33,8 @@ along with qpdfview.  If not, see <http://www.gnu.org/licenses/>.
+ 
+ #include <QtWidgets>
+ #include <QtPrintSupport>
+-#include <poppler-qt5.h>
```

Comment:
This include is actually correct, but the `c{,xx}flags` has to be populated with the output of `pkg-config --cflags poppler-qt5`. I think you should instead patch the `.pro` file to make use of `pkg-config`.

Reply:
i'm sorry @alaviss, i'm not really understand that pkg-config thing.. do you mean i'm just delete that  row and add on .pro on haiku part?

Reply:
I think what @alaviss suggests is creating a `PATCH()` function in the recipe with something that might look like:
```
PATCH()
{
	sed -i \
		-e "/INCLUDEPATH/ s|/usr/include/poppler/qt4|`pkg-config --cflags poppler-qt4|`" \
		-e "/INCLUDEPATH/ s|-L/usr/lib|`pkg-config --cflags poppler-qt5|`" \
		-e "/LIBS/ s|/usr/include/poppler/qt4|`pkg-config --libs-only-L poppler-qt4|`" \
		-e "/LIBS/ s|/usr/include/poppler/qt5|`pkg-config --libs-only-L poppler-qt5|`" \
		booksorganizer.pro
}
```
Warning: I guess you will only need to use half of the `-e` patterns: the ones for either Qt4 or Qt5.
BTW, you can rework this to have shorter lines, for instance, by first defining e.g. qt5includeDir and qt5libDir and then using their values in the sed commands.

Reply:
whoa.. sed thing..  i need to learn that before use it...
btw.. i've been create new patch..  based on @alaviss  suggestion.. 
it change location where to save Books db and thumbnails.. 
please see again @fbrosson  and @Begasus ..

Reply:
@fbrosson  if that @alaviss  want.. can i ask help to you @fbrosson  to accomodate @alaviss  request?

Reply:
I haven't tested the recipe with your new patchset, but I can see that the patchset can be simplified because in the first patch chunk you do some changes to `bookinfo.cpp`, but later, in another patch chunk, you remove what the first patch chunk did.
One technique to simplify the patchset is to go in `/boot/home/haikuports/app-text/booksorg/work-0.3/sources/booksorg_03/` and use git to squash some (or all) of the commits, then check that `git log -p` is fine and then regenerate the patchset with `hp -e booksorg`.

Below are the patch chunks that could be squashed. Others may also be squashed, but it's better that you keep unrelated changes in separate patches (of the same patchset file, of course.)
```diff
From 5d7fe1ede47e9f2f49c2b161050b13762e7dd447 Mon Sep 17 00:00:00 2001
From: mazbrili <mazbrili@yahoo.com>
Date: Sat, 20 Oct 2018 18:56:36 +0700
Subject: apply change


diff --git a/bookinfo.cpp b/bookinfo.cpp
old mode 100644
new mode 100755
index f75bcaa..1c69276
--- a/bookinfo.cpp
+++ b/bookinfo.cpp
@@ -20,7 +20,7 @@
  */
 #include "bookinfo.h"
 #if QT_VERSION >= 0x050000
-#include <poppler-qt5.h>
+#include <poppler/qt5/poppler-qt5.h>
 #else
 #include <poppler-qt4.h>
#endif
```

```diff
From 0e16d9e3c17d6787c15551b2b0c04dcc9b51d6dd Mon Sep 17 00:00:00 2001
From: mazbrili <mazbrili@yahoo.com>
Date: Mon, 29 Oct 2018 20:59:46 +0700
Subject: change path with builtin QStandardPaths from Qt


diff --git a/bookinfo.cpp b/bookinfo.cpp
index bfac9c1..35a7d3e 100755
--- a/bookinfo.cpp
+++ b/bookinfo.cpp
@@ -19,17 +19,15 @@
  * Boston, MA 02110-1301, USA.
  */
 #include "bookinfo.h"
-#if QT_VERSION >= 0x050000
-#include <poppler/qt5/poppler-qt5.h>
-#else
-#include <poppler-qt4.h>
-#endif
```

Reply:
@mazbrili 

You'd need to put these somewhere in the `.pro` file:

```
CONFIG += link_pkgconfig
PKGCONFIG += poppler-qt5
```

Qt has a lot of features ;)

Reply:
i try to wipe all #include <poppler/qt5/poppler-qt5.h> but cannot compile anymore..  complained about undeclared..blah blah poppler thing..

Reply:
@alaviss please review latest update.. sorry i  still cannot use pkgconfig thing...

Reply:
No worries, I'll try it myself and let you know if I found a fix ;)

Reply:
For some reason it works here :/ I've reversed all of the poppler patch and tested on my machine.

Could you try `#include <poppler-qt5.h>` again?

Reply:
Done.

@fbrosson @mazbrili could you test my changes?

Reply:
@alaviss: Excellent! I checked with your update (a8abb5b46c0c98b18d2a9ce17a30099faf6f3413) before merging, and it worked like a charm!

Reply:
> Done.
> 
> @fbrosson @mazbrili could you test my changes?

sorry... that time seem i'm still sleep... thanks anyway